src/AppInstallerCLICore/Workflows/PortableInstallFlow.cpp

```diff
@@ -0,0 +1,396 @@
+                AICLI_LOG(CLI, Verbose, << "Writing to Uninstall registry complete.");
+            }
+
+            return uninstallEntry;
+        }
+    }
+
+    void EnsureFeatureEnabledForPortableInstall(Execution::Context& context)
+    {
+        auto installerType = context.Get<Execution::Data::Installer>().value().InstallerType;
+
+        if (installerType == InstallerTypeEnum::Portable)
+        {
+            context << Workflow::EnsureFeatureEnabled(Settings::ExperimentalFeature::Feature::PortableInstall);
+        }
+    }
+
+    void EnsureNonReservedNamesForPortableInstall(Execution::Context& context)
+    {
+        std::filesystem::path targetFullPath = GetPortableTargetFullPath(context);
+        std::filesystem::path symlinkFullPath = GetPortableSymlinkFullPath(context);
+        if (!Filesystem::IsNonReservedFilename(targetFullPath) || !Filesystem::IsNonReservedFilename(symlinkFullPath))
+        {
+            context.Reporter.Error() << Resource::String::ReservedFilenameError << std::endl;
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_PORTABLE_INSTALL_FAILED);
+        }
+    }
+
+    void PortableInstall(Execution::Context& context)
+    {
+        context.Reporter.Info() << Resource::String::InstallFlowStartingPackageInstall << std::endl;
+
+        try
+        {
+            Registry::Key entry = WritePortableEntryToUninstallRegistry(context);
+            const std::filesystem::path installerPath = context.Get<Execution::Data::InstallerPath>();
+            std::filesystem::path targetFullPath = ConvertToUTF16(entry[Normalize(s_PortableTargetFullPath)].value().GetValue<Value::Type::String>());
+            std::filesystem::path symlinkFullPath = ConvertToUTF16(entry[Normalize(s_PortableSymlinkFullPath)].value().GetValue<Value::Type::String>());
+            std::filesystem::path linksDirectoryPath = ConvertToUTF16(entry[Normalize(s_PortableLinksDirectory)].value().GetValue<Value::Type::String>());
+
+            std::filesystem::rename(installerPath, targetFullPath);
+            AICLI_LOG(CLI, Info, << "Portable exe moved to: " << targetFullPath);
+
+            if (std::filesystem::is_symlink(symlinkFullPath))
+            {
+                std::filesystem::remove(symlinkFullPath);
+            }
+
+            if (std::filesystem::exists(symlinkFullPath))
+            {
+                std::filesystem::remove(symlinkFullPath);
+            }
+
+            if (AppInstaller::Filesystem::SupportsReparsePoints(targetFullPath))
+            {
+                std::filesystem::create_symlink(targetFullPath, symlinkFullPath);
+                AICLI_LOG(CLI, Info, << "Portable symlink created at: " << symlinkFullPath);
+            }
+            else
+            {
+                context.Reporter.Error() << Resource::String::ReparsePointsNotSupportedError << std::endl;
+                AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_PORTABLE_REPARSE_POINT_NOT_SUPPORTED);

```

Comment:
We create a symlink for every portable package and place them in a symlinks directory that is added to path once. This is our solution to minimize the number of times we need to append to path while still being able to support adding to path and running the tool from the command line. If we are unable to create the symlink, then portable installation will have failed since the user won't be able to run their tool directly from the command line.

Reply:
But the path to the binaries could be added to PATH instead of the path to the symlink?

Reply:
I see; I thought that each package was added to path separately. If symlinks aren't supported, I would argue that the binaries should still be added to the path though, even if it does result in multiple path modifications

Reply:
There are limitations to the length of the Path, and we've seen performance problems when the path becomes arbitrarily long. With the symbolic link directory in the path, users can run the commands from any path in their terminal.

Reply:
Interesting. I thought that the limits were in the many thousands of characters and wouldn’t be an issue, especially since environment variables can be chained

But, it does make sense that it would impact performance. Thank you for the explanation!